src/Halogen/Menu/Component.purs

```diff
@@ -0,0 +1,124 @@
+module Halogen.Menu.Component where
```

Comment:
Cool thats a good distinction. States may be defined in terms of re-usable models but probably aren't re-useable models themselves. Are you using the `State` suffix for these data values / types?